JPA Annotation Processing problems in Eclipse Oxygen 3a

As requested in #2274 this is a new issue for the annotation processing problems with QueryDSL 4.1.4 and 4.2.1 in Eclipse Oxygen 3a:

````
java.lang.Exception: java.lang.IllegalArgumentException: Executable, package and module are illegal argument for Types.contains(..)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:169)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:121)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:135)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:933)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:443)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:419)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:372)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:187)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:305)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:61)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:256)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:186)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.resources.Workspace.buildInternal(Workspace.java:487)
	at org.eclipse.core.internal.resources.Workspace.build(Workspace.java:399)
	at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$5.run(SetupTaskPerformer.java:3369)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: java.lang.IllegalArgumentException: Executable, package and module are illegal argument for Types.contains(..)
	at org.eclipse.jdt.internal.compiler.apt.model.TypesImpl.validateRealType(TypesImpl.java:228)
	at org.eclipse.jdt.internal.compiler.apt.model.TypesImpl.erasure(TypesImpl.java:285)
	at com.querydsl.apt.jpa.JPAConfiguration.getRealElementType(JPAConfiguration.java:116)
	at com.querydsl.apt.jpa.JPAConfiguration.getRealType(JPAConfiguration.java:93)
	at com.querydsl.apt.TypeElementHandler.handleEntityType(TypeElementHandler.java:123)
	at com.querydsl.apt.AbstractQuerydslProcessor.processAnnotations(AbstractQuerydslProcessor.java:117)
	at com.querydsl.apt.AbstractQuerydslProcessor.process(AbstractQuerydslProcessor.java:89)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:139)
	... 25 more
````

--- a/querydsl-apt/src/main/java/com/querydsl/apt/jpa/JPAConfiguration.java
+++ b/querydsl-apt/src/main/java/com/querydsl/apt/jpa/JPAConfiguration.java
@@ -113,7 +113,12 @@ public class JPAConfiguration extends DefaultConfiguration {
         }
         if (mirror != null) {
             TypeMirror typeArg = TypeUtils.getAnnotationValueAsTypeMirror(mirror, "targetEntity");
-            TypeMirror erasure = types.erasure(element.asType());
+            TypeMirror erasure;
+            if (element instanceof ExecutableElement) {
+                erasure = ((ExecutableElement) element).getReturnType();
+            } else {
+                erasure = types.erasure(element.asType());
+            }
             TypeElement typeElement = (TypeElement) types.asElement(erasure);
             if (typeElement != null && typeArg != null) {
                 if (typeElement.getTypeParameters().size() == 1) {

